src/core/service.c

```diff
@@ -552,6 +552,11 @@ static int service_verify(Service *s) {
         if (UNIT(s)->load_state != UNIT_LOADED)
                 return 0;
 
+        if (!service_unit_name_is_valid(UNIT(s)->id)) {
+                log_unit_error(UNIT(s), "Service name is invalid or reserved. Refusing.");
+                return -ENOEXEC;
```

Comment:
EINVAL seems more appropriate, we usually use ENOEXEC in this kind of context to mean somehow `Exec*` is noncompliant